controllers/argocd/metrics.go

```diff
@@ -0,0 +1,147 @@
+package argocd
```

Comment:
How about creating a new package for this ? 
`package metrics`

Reply:
Sure
I figured we have to do it as part of the refactor eventually anyway

Reply:
I am ok with it